packages/gatsby/src/bootstrap/load-themes/index.js

```diff
@@ -5,25 +5,46 @@ const _ = require(`lodash`)
 const debug = require(`debug`)(`gatsby:load-themes`)
 const preferDefault = require(`../prefer-default`)
 const getConfigFile = require(`../get-config-file`)
+const reporter = require(`gatsby-cli/lib/reporter`)
 
 // get the gatsby-config file for a theme
-const resolveTheme = async themeSpec => {
+const resolveTheme = async (themeSpec, configFileThatDeclaredTheme) => {
   const themeName = themeSpec.resolve || themeSpec
   let themeDir
   try {
+    // theme is an node-resolvable module
     themeDir = path.dirname(require.resolve(themeName))
   } catch (e) {
-    // this can be local plugin, and require.resolve will throw
-    // in this case - let's return partial entry
-    return { themeName, themeSpec }
+    // is a local plugin OR it doesn't exist
+    const pathToLocalTheme = path.join(path.resolve(`.`), `plugins`, themeName)
```

Comment:
we always resolve to main site local plugin which results in error like this (if plugin declared by some theme can't be found):
```
Couldn't find "wat" plugin declared in "/Users/misiek/dev/gatsby/e2e-tests/themes/gatsby-theme-about/gatsby-config.js".

Tried looking for a local plugin in /Users/misiek/dev/gatsby/e2e-tests/themes/production-runtime/plugins/wat
```
So plugin is declared in theme but we look for it in main site local plugins.

This is very confusing - I think we should not even look for local plugin in those cases, and adjust error message (on it right now)